### ISSUE TITLE
vega normal maps: which direction?

### ISSUE THREAD

**Issue by illwieckz** [UnvanquishedAssets/tex-vega_src.dpkdir#1]
While investigating [parallax mapping](https://github.com/DaemonEngine/Daemon/issues/30), I discovered the fact [DirectX and OpenGL normalmaps are reverted together](https://answers.arnoldrenderer.com/questions/169/why-do-my-normal-maps-appear-to-be-inverted.html).

@IngarKCT, do you know if you intended to do the Vega normal maps the DirectX way or the OpenGL way or did you intended nothing?

**Comment by IngarKCT** [comment]
I just wanted it to work :-) I assume they need to be OpenGL compatible. Is there a technical description of what kind of maps the engine supports/requires? (normal maps or others)

**Comment by illwieckz** [comment]
Fomat of normal maps shipped within this repository (same with `map-vega` one) is `-X -Y Z` which is very exotic…

To prevent rewriting the git history (and to make things easier for similar case), the `normalFormat` material keyword was added to the engine, so I just added `normalFormat -X -Y Z` to the `.shader` files and everything looks good.

This issue is fixed by 91632b622761e129537288069f6be82e4c313635.

For reference, DirectX format is `X -Y Z` and OpenGL format is `X Y Z`.

Without other instruction (like using `normalFormat` keyword), the Daemon Engine assumes normal format to be in DirectX format (`X -Y Z`) when reading Quake3-like `.shader` files because that's the format used by Doom3 and XreaL mimicked which was then mimicked by almost all idTech3-derivated engines.

To help recognise files, with DirectX format, light seems to come from the floor, while with OpenGL format, light seems to come from the sky (which feels more natural).
